Service/src/main/java/service/productParser/CelParser.java

```
@@ -0,0 +1,45 @@
+package service.productParser;
+
+import domain.WishlistItem;
+import domain.WishlistItemVendor;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CelParser implements ProductParser {
+    private static final String CEL_URL_SEARCH = "https://www.cel.ro/cauta/:keyword";
+
+    @Override
+    public List<WishlistItem> getProductsByKeyword(String keyword) {
+        try {
+            // CEL substitutes every space in the search with a plus sign in its URL
+            keyword = keyword.replaceAll(" ", "+");
+            Document document = Jsoup.connect(CEL_URL_SEARCH.replaceAll(":keyword", keyword)).get();
```

Not a blocker for merge, but I think a whitelist of characters might help here (_blacklists don't work_). I suggest a whitelist with only alphanumeric characters (maybe a dot and underscore also, but they are not essential). As sending an error to the front end for invalid characters in a search field may be unpleasant, silently dropping non-whitelisted characters and moving on with the search should be fine.